controllers/managedocs_controller.go

```diff
@@ -1786,3 +1795,25 @@ func (r *ManagedOCSReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDeviceS
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *ManagedOCSReconciler) reconcileRookOverrideConfig() error {
+	if r.DeploymentType != providerDeploymentType {
+		r.Log.Info("Non provider deployment, skipping reconcile for rook override config")
+		return nil
+	}
+	r.Log.Info("Reconciling Rook Override Config Map")
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.rookConfigOverride, func() error {
+		if err := r.own(r.rookConfigOverride); err != nil {
+			return err
+		}
+
+		r.rookConfigOverride.Data = map[string]string{
+			"config": "[global]\nosd_pool_pg_autoscale_mode = off\nosd_pool_default_pg_num = 128\nosd_pool_default_pgp_num = 128",
```

Comment:
This seems problematic to read and change in the future, can we just do the following instead?
```suggestion
			"config": "" +
				"[global]\n" +
				"osd_pool_pg_autoscale_mode = off\n" +
				"osd_pool_default_pg_num = 128\n" +
				"osd_pool_default_pgp_num = 128\n",
```

Reply:
Done